src/Arbitrary.hs

```diff
@@ -804,7 +807,9 @@ instance Aeson.ToJSON D3DN where
 -- * event log
 
 instance Arbitrary EventLog where
-    arbitrary = garbitrary
+    arbitrary = EventLog <$> arbWord <*> nonEmpty
+      where
+        nonEmpty = (:) <$> garbitrary <*> garbitrary
```

Comment:
my guess is that this is an instance of "let should not be generalized". Adding a type sig should fix it.